plugins/vpp/vpp.go

```diff
@@ -56,17 +56,32 @@ func FindCompatibleBinapi(ch CompatibilityChecker) (binapi.Version, error) {
 	if len(binapi.Versions) == 0 {
 		return "", fmt.Errorf("no binapi versions loaded")
 	}
-	logging.Debugf("checking compatibility for %d binapi versions", len(binapi.Versions))
+	logging.Debugf("checking binapi compatibility (%d versions)", len(binapi.Versions))
+
+	var mostCompatible = struct {
+		version      Version
+		incompatible int
+	}{}
 	for version, msgList := range binapi.Versions {
 		msgs := msgList.AllMessages()
 		if err := ch.CheckCompatiblity(msgs...); err == nil {
 			logging.Debugf("found compatible binapi version: %v", version)
 			return version, nil
 		} else if ierr, ok := err.(*govppapi.CompatibilityError); ok {
-			logging.Debugf("binapi version %-15v incompatible: %d/%d incompatible messages", version, len(ierr.IncompatibleMessages), len(msgs))
+			logging.Debugf("binapi version %-15v incompatible: %d/%d incompatible messages",
+				version, len(ierr.IncompatibleMessages), len(msgs))
+
+			if mostCompatible.version == "" || mostCompatible.incompatible > len(ierr.IncompatibleMessages) {
+				mostCompatible.version = version
```

Comment:
finally I've found something. `mostCompatible` can be version without compatible messages. Please tell me that I'm right and it is something :)

Reply:
Hm.. actually I think you're on point! I should check if at least some messages are compatible.